pin8, new recipe

cnelmortimer opened the pull request:
Dear repository masters,

Please find the HaikuPorts recipe for Pin8, a native Chip-8 emulator for Haiku OS written in C++.

Kind regards.

diversys commented:
Please take a look at recipe formatting https://github.com/haikuports/haikuporter/blob/master/generic/generic_cmd-1.2.3.recipe
When you fix your recipe there is no need to close this PR and open new one. 
You can just force push to update this PR using: `git push -f`

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,49 @@
+SUMMARY="A Chip-8 emulator for Haiku OS"
+
+DESCRIPTION="This is a Chip-8 emulator for Haiku OS. Hence, it allows you to run Chip-8 games and software. Blitz, Blinky and Tetris are some games available for this platform. A sample game, a flappy bird clone, is also available in the webpage of this program. \ Pin8 features responsive design, real keyboard compatibility, sound and two running modes."
```
Wrap at 80th character. BTW, it's just Haiku, not Haiku OS

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,49 @@
+
+DESCRIPTION="This is a Chip-8 emulator for Haiku OS. Hence, it allows you to run Chip-8 games and software. Blitz, Blinky and Tetris are some games available for this platform. A sample game, a flappy bird clone, is also available in the webpage of this program. \ Pin8 features responsive design, real keyboard compatibility, sound and two running modes."
+
+HOMEPAGE="https://github.com/cnelmortimer/Pin8/blob/master/README.md"
```
https://github.com/cnelmortimer/Pin8

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,49 @@
+
+SOURCE_DIR="Pin8-$srcGitRev"
+
+ARCHITECTURES="x86_gcc2 x86 ?x86_64"
```
We dropped x86 ARCH recently.

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,49 @@
+ARCHITECTURES="x86_gcc2 x86 ?x86_64"
+
+PROVIDES="
+	Pin8 = $portVersion
```
Lower case in this case

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,49 @@
+	
+BUILD()
+{
+	make	
```
Trailing whitespace

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,49 @@
+{
+	mkdir -p $appsDir
+	cp -a Pin8 $appsDir
+	addAppDeskbarSymlink $appsDir/Pin8	
```
Trailing whitespace

pulkomandy commented:
@diver: the x86 arch is still there, we just stopped distributing nightlies. It's still possible for people to build their own Haiku images if they wish so, and I don't see a reason for removing support yet?

cnelmortimer commented:
Ok @diversys , I think I have done what you required (or at least, I really hope so... ). Sorry for any inconvenience and thank you for the guidelines.

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,43 @@
+this platform. A sample game, a flappy bird clone, is also available in the \
+webpage of this program. \Pin8 features responsive design, real keyboard \
+compatibility, sound and two running modes."
+HOMEPAGE="https://github.com/cnelmortimer/Pin8/blob/master/README.md"
```
Should be HOMEPAGE="https://github.com/cnelmortimer/Pin8"

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,43 @@
+DESCRIPTION="This is a Chip-8 emulator. Hence, it allows you to run Chip-8 \
+games  and software. Blitz, Blinky and Tetris are some games available for \
+this platform. A sample game, a flappy bird clone, is also available in the \
+webpage of this program. \Pin8 features responsive design, real keyboard \
```
Extra slash? \Pin8

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,43 @@
+SUMMARY="A Chip-8 emulator"
+DESCRIPTION="This is a Chip-8 emulator. Hence, it allows you to run Chip-8 \
+games  and software. Blitz, Blinky and Tetris are some games available for \
```
extra space "games  and"

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,43 @@
+
+PROVIDES="
+	pin8 = $portVersion
+	app:pin8 = $portVersion
```
app:Pin8 should be uppercased (unlike the previous line which is needs to be always lowercase)

diversys commented on games-emulation/pin8/pin8-1.0.recipe:
```diff
@@ -0,0 +1,43 @@
+	haiku_devel
+	cmd:g++
+	cmd:make
+	cmd:xres
```
cmd:xres is provided by haiku package, I think it can be dropped here.

pulkomandy replied:
in fact it is a good idea to declare it. If we ever decide to split the haiku package in smaller parts, or move the command to another package, we don't need to update any recipes then. That's the point of all these cmd: and lib: provides, and I see no reason to make an exception for the haiku package.

diversys replied:
Noted

cnelmortimer commented:
Hi again @diversys , 

I have removed the extra space after game, the extra slash before Pin8, cropped the homepage, uppercased "app:Pin8" and removed the "cmd:xres" dependency. Many thanks for your patience.
Kind regards.
